src/main/java/seedu/address/model/studyspot/OperatingHours.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.model.studyspot;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a StudySpot's operating hours in the study tracker.
+ * Guarantees: immutable; is valid as declared in {@link #isValidOperatingHours(String)}
+ */
+public class OperatingHours {
+
+    public static final String MESSAGE_CONSTRAINTS = "Operating hours are in the format: "
+            + "[HHmm-HHmm, HHmm-HHmm], "
+            + "where the first argument represents operating hours in the weekdays, "
+            + "while the second argument represents operating hours in the weekends.";
+    public static final String DEFAULT_OPERATING_HOURS = "unspecified";
+
+    /** Ensures the time provided is valid and in the format hh:mm. */
+    public static final String TIME_VALIDATION_REGEX = "([01]?[0-9]|2[0-3])[0-5][0-9]";
+
+    public final String value;
+    public final String weekdaysValue;
+    public final String weekendsValue;
+
+    /**
+     * Constructs an {@code OperatingHours}.
+     *
+     * @param operatingHours A valid operating hours.
+     */
+    public OperatingHours(String operatingHours) {
+        requireNonNull(operatingHours);
+        checkArgument(isValidOperatingHours(operatingHours), MESSAGE_CONSTRAINTS);
+
+        value = operatingHours;
+
+        if (operatingHours.equals(DEFAULT_OPERATING_HOURS)) {
+            weekdaysValue = DEFAULT_OPERATING_HOURS;
+            weekendsValue = DEFAULT_OPERATING_HOURS;
+        } else {
+            String[] separatedOperatingHours = operatingHours.split(",", 2);
+            weekdaysValue = separatedOperatingHours[0].trim();
+            weekendsValue = separatedOperatingHours[1].trim();
+        }
+
+    }
+
+    /**
+     * Constructs an OperatingHours with no value.
+     */
+    private OperatingHours() {
+        value = DEFAULT_OPERATING_HOURS;
+        weekdaysValue = DEFAULT_OPERATING_HOURS;
+        weekendsValue = DEFAULT_OPERATING_HOURS;
+    }
+
+    /**
+     * Returns if a given string is a valid operating hours.
+     */
+    public static boolean isValidOperatingHours(String test) {
+        if (test.equals(DEFAULT_OPERATING_HOURS)) {
+            return true;
+        }
+
+        String[] separatedOperatingHours = test.split(",", 2);
+        if (separatedOperatingHours.length == 1) {
+            return false;
+        }
+        return isValidOperatingHour(separatedOperatingHours[0].trim())
+                && isValidOperatingHour(separatedOperatingHours[1].trim());
+    }
+
+    private static boolean isValidOperatingHour(String test) {
+        String[] separatedOperatingHour = test.split("-", 2);
+        if (separatedOperatingHour.length == 1) {
+            return false;
+        }
+        return separatedOperatingHour[0].trim().matches(TIME_VALIDATION_REGEX)
+                && separatedOperatingHour[1].trim().matches(TIME_VALIDATION_REGEX);
+    }
+
```

Comment:
We also should check if `OperatingHour[0]` < `OperatingHour[1]`, otherwise we have times like `1800-0900`